src/devices/infactory.c

```diff
@@ -1,84 +1,127 @@
     the Free Software Foundation; either version 2 of the License, or
     (at your option) any later version.
 */
 /**
-inFactory Outdoor sensor transmits data temperature, humidity.
+Outdoor sensor, transmits temperature and humidity data
+- inFactory NC-3982-913/NX-5817-902, Pearl (for FWS-686 station)
+- nor-tec 73383 (weather station + sensor), Schou Company AS, Denmark
+- DAY 73365 (weather station + sensor), Schou Company AS, Denmark
 
-Also NC-3982-913 from Pearl (for FWS-686 station).
 
-Transmissions also includes an id. The sensor transmits
-every 60 seconds 6 packets.
+Transmissions also includes an id. Every 60 seconds
+the sensor transmits 6 packets:
 
     0000 1111 | 0011 0000 | 0101 1100 | 1110 0111 | 0110 0001
-    xxxx xxxx | cccc cccc | tttt tttt | tttt hhhh | hhhh ??nn
-
-- x: ID // changes on battery switch
-- c: Unknown Checksum (changes on every transmit if the other values are different)
-- h: Humidity // BCD-encoded, each nibble is one digit
-- t: Temperature   // in °F as binary number with one decimal place + 90 °F offset
+    iiii iiii | cccc ub?? | tttt tttt | tttt hhhh | hhhh ??nn
+
+- i: identification // changes on battery switch
+- c: CRC-4 // CCITT checksum, see below for computation specifics
+- u: unknown // (sometimes set at power-on, but not always)
+- b: battery low // flag to indicate low battery voltage
+- h: Humidity // BCD-encoded, each nibble is one digit, 'A0' means 100%rH
+- t: Temperature // in °F as binary number with one decimal place + 90 °F offset
 - n: Channel // Channel number 1 - 3
 
 Usage:
 
     # rtl_433 -f 434052000 -R 91 -F json:log.json
+    # rtl_433 -R 91 -F json:log.json
+    # rtl_433 -C si
 
 Payload looks like this:
 
-    [00] { 4} 00             : 0000
-    [01] {40} 0f 30 5c e7 61 : 00001111 00110000 01011100 11100111 01100001
+    [00] {40} 0f 30 5c e7 61 : 00001111 00110000 01011100 11100111 01100001
 
-First row is actually the preamble part. This is added to make the signal more unique.
+(See below for more information about the signal timing.)
 */
 
 #include "decoder.h"
 
+static void infactory_raw_msg(char *msg, uint8_t *b) {
+    int i;
+    char c;
+    char *p = msg;
+    for (i = 0; i < 10; i++) {
+        c = ((b[i >> 1]) >> (4 * (1 - (i & 1)))) & 0x0F;
+        c = (c > 9) ? c + 55 : c + 48;
+        *p++ = c;
+    }
+    *p = 0;
+}
+
+static int infactory_crc_check(uint8_t *b) {
+    int i;
+    uint8_t msg_crc, crc;
+    uint8_t msg[5];
+    memcpy(msg, b, 5);
+    msg_crc = msg[1] >> 4;
+    // for CRC computation, channel bits are at the CRC position(!)
+    msg[1] = (msg[1] & 0x0F) | (msg[4] & 0x0F) << 4;
+    crc = msg[0] >> 4;
+    for (i = 4; i < 36; i++) {
+        crc <<= 1;
+        if (msg[i>>3] & (0x80 >> (i&7))) {
+            crc |= 1; 
+        }
+        if (crc & 0x10) {
+            crc ^= 0x13; // Koopmann 0x9, CCITT-4; FP-4; ITU-T G.704
+        }
+    }
+
+    return (crc == msg_crc);
+}
+
```

Comment:
https://github.com/merbanan/rtl_433/blob/master/src/util.c#L64 can this be used?

Reply:
Unfortunately not - the CRC is computed over 36 message bits, but crc4() only computes over full bytes.

Reply:
CRC is invariant if you prepend zero bits. I'll have a closer look an give an example tomorrow

Reply:
I found a way to compute the CRC using crc4(). :-)